lib/m2ts/m2ts.js

```diff
@@ -310,51 +310,58 @@ ElementaryStream = function() {
     programMapTable,
     parsePes = function(payload, pes) {
       var ptsDtsFlags;
-
-      // get the packet length, this will be 0 for video
-      pes.packetLength = 6 + ((payload[4] << 8) | payload[5]);
-
-      // find out if this packets starts a new keyframe
-      pes.dataAlignmentIndicator = (payload[6] & 0x04) !== 0;
-      // PES packets may be annotated with a PTS value, or a PTS value
-      // and a DTS value. Determine what combination of values is
-      // available to work with.
-      ptsDtsFlags = payload[7];
-
-      // PTS and DTS are normally stored as a 33-bit number.  Javascript
-      // performs all bitwise operations on 32-bit integers but javascript
-      // supports a much greater range (52-bits) of integer using standard
-      // mathematical operations.
-      // We construct a 31-bit value using bitwise operators over the 31
-      // most significant bits and then multiply by 4 (equal to a left-shift
-      // of 2) before we add the final 2 least significant bits of the
-      // timestamp (equal to an OR.)
-      if (ptsDtsFlags & 0xC0) {
-        // the PTS and DTS are not written out directly. For information
-        // on how they are encoded, see
-        // http://dvd.sourceforge.net/dvdinfo/pes-hdr.html
-        pes.pts = (payload[9] & 0x0E) << 27 |
-          (payload[10] & 0xFF) << 20 |
-          (payload[11] & 0xFE) << 12 |
-          (payload[12] & 0xFF) <<  5 |
-          (payload[13] & 0xFE) >>>  3;
-        pes.pts *= 4; // Left shift by 2
-        pes.pts += (payload[13] & 0x06) >>> 1; // OR by the two LSBs
-        pes.dts = pes.pts;
-        if (ptsDtsFlags & 0x40) {
-          pes.dts = (payload[14] & 0x0E) << 27 |
-            (payload[15] & 0xFF) << 20 |
-            (payload[16] & 0xFE) << 12 |
-            (payload[17] & 0xFF) << 5 |
-            (payload[18] & 0xFE) >>> 3;
-          pes.dts *= 4; // Left shift by 2
-          pes.dts += (payload[18] & 0x06) >>> 1; // OR by the two LSBs
+      const startPrefix = payload[0] << 16 | payload[1] << 8 | payload[2]
+      // default to an empty array
+      pes.data = new Uint8Array()
+      // In certain live streams, the start of a TS fragment has ts packets
+      // that are frame data that is continuing from the previous fragment. This
+      // is to check that the pes data is the start of a new pes payload
+      if (startPrefix === 1) {
```

Comment:
can this be changed to be
```js
if (startPrefix !== 1) {
 return;
}
```
That way the current code doesn't need another level of indentation?

Reply:
Sure